### PR TITLE
Disable some addons for April Fools

### PR DESCRIPTION
**Pull request by WorldLanguages**
See #5848 from 2023

### CODE REVIEW

**Comment by WorldLanguages** [comment]
Vanilla Scratch with `"everything_is_totally_normal": true` is also quite buggy.

## THE FOLLOWING VIDEOS ARE VANILLA SCRATCH

brave_yEHrYIBidE
brave_obvRsDABWx

**Comment by Samq64** [comment]
Nothing changes when setting the timestamp to the same one you said on #7299.

**Comment by WorldLanguages** [comment]
> Nothing changes when setting the timestamp to the same one you said on #7299.

Changing the system time only takes effect in this PR if Redux is unavailable. Try making content-scripts/load-redux.js empty or broken.

File: addons/hide-flyout/april-fools.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const isScratchAprilFools24 = async (redux) => {
+  let val = false;
```
Comment (WorldLanguages) [inline addons/hide-flyout/april-fools.js:2]: Should have added
`ìf(april_three_or_later) return false;`

File: addons/hide-flyout/april-fools.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  const now = Date.now() / 1000;
+  if (!redux.state && now < 1712059200 && now > 1711886400) val = true;
+  else {
+    await redux.waitForState((state) => state.session?.session?.flags);
```
Comment (WorldLanguages) [inline addons/hide-flyout/april-fools.js:6]: This didn't work 100% of the time on runAtComplete:false

**Pull request merged**